backend/src/election/mod.rs

```diff
@@ -103,17 +106,163 @@ pub async fn election_status(
+            self.election.category.to_eml_code(),
+            self.election.election_date.year(),
+            self.election.location.replace(" ", "_"),
+        )
+    }
+
+    fn zip_filename(&self) -> String {
+        use chrono::Datelike;
+        format!(
+            "election_result_{}{}_{}.zip",
+            self.election.category.to_eml_code(),
+            self.election.election_date.year(),
+            self.election.location.replace(" ", "_"),
+        )
+    }
+
+    fn into_pdf_model(self, xml_hash: impl Into<String>) -> PdfModel {
+        PdfModel::ModelNa31_2(ModelNa31_2Input {
+            polling_stations: self.polling_stations,
+            summary: self.summary,
+            election: self.election,
+            hash: xml_hash.into(),
+            creation_date_time: self.creation_date_time.format("%d-%m-%Y %H:%M").to_string(),
+        })
+    }
+}
+
+/// Download a zip containing a PDF for the PV and the EML with election results
+#[utoipa::path(
+    get,
+    path = "/api/elections/{election_id}/download_zip_results",
+    responses(
+        (
+            status = 200,
+            description = "ZIP",
+            content_type = "application/zip",
+            headers(
+                ("Content-Disposition", description = "attachment; filename=\"filename.zip\"")
+            )
+        ),
+        (status = 404, description = "Not found", body = ErrorResponse),
+        (status = 500, description = "Internal server error", body = ErrorResponse),
+    ),
+    params(
+        ("election_id" = u32, description = "Election database id"),
+    ),
+)]
+pub async fn election_download_zip_results(
+    State(elections_repo): State<Elections>,
+    State(polling_stations_repo): State<PollingStations>,
+    State(polling_station_results_entries_repo): State<PollingStationResultsEntries>,
+    Path(id): Path<u32>,
+) -> Result<Attachment<Vec<u8>>, APIError> {
+    use std::io::Write;
+
+    let input = ResultsInput::new(
+        id,
+        elections_repo,
+        polling_stations_repo,
+        polling_station_results_entries_repo,
+    )
+    .await?;
+    let xml = input.as_xml();
+    let xml_string = xml.to_xml_string()?;
+    let pdf_filename = input.pdf_filename();
+    let xml_filename = input.xml_filename();
+    let zip_filename = input.zip_filename();
+    let model = input.into_pdf_model(eml_document_hash(&xml_string, true));
+    let content = generate_pdf(model)?;
+
+    let mut buf = vec![];
+    let mut cursor = std::io::Cursor::new(&mut buf);
+    let mut zip = zip::ZipWriter::new(&mut cursor);
+    let options = SimpleFileOptions::default()
+        .compression_method(zip::CompressionMethod::DEFLATE)
+        // zip file format does not support dates beyond 2107 or inserted leap (i.e. 61th) second
+        .last_modified_time(
+            chrono::Local::now()
+                .naive_local()
+                .try_into()
+                .expect("Timestamp outside of zip range"),
```

Comment:
I think the [recommended message style](https://doc.rust-lang.org/std/result/enum.Result.html#recommended-message-style) is something like:
```suggestion
                .expect("Timestamp should be inside zip range"),
```